pydantic/root_model.py

```diff
@@ -25,6 +27,7 @@
 RootModelRootType = typing.TypeVar('RootModelRootType')
 
 
+@dataclass_transform(kw_only_default=False, field_specifiers=(PydanticModelField,))
 class RootModel(BaseModel, typing.Generic[RootModelRootType]):
     """Usage docs: https://docs.pydantic.dev/2.6/concepts/models/#rootmodel-and-custom-root-types
```

Comment:
Although this works with `mypy` (e.g. creating subclass of `RootModel` will allow positional argument for `root`), pyright does not support it (see: https://github.com/microsoft/pyright/issues/6489#issuecomment-1817642695).

One workaround would be to have a metaclass for `RootModel`, subclassing `ModelMetaclass`, and we could set `dataclass_transform.kw_only_default=False` on this metaclass.

A fix to support having `dataclass_transform` applied directly on `RootModel` is also available, so this is up to you (see issue on pyright for more context).